aquadoggo/src/materializer/tasks/dependency.rs

```diff
@@ -654,4 +722,376 @@ mod tests {
             assert_eq!(schema_tasks, expected_schema_tasks);
         });
     }
+
+    #[rstest]
+    fn materialise_late_views(key_pair: KeyPair) {
```

Comment:
Haha, epic test :rofl: love it.

Reply:
Hehe, it has a lot of boilerplate because I needed to enforce the order of things being processed without also adding the whole replication system 🙈